p2p/src/sessions/mod.rs

```diff
@@ -47,41 +53,69 @@ impl<T> Default for Sessions<T> {
         Self {
             server_address: None,
             inbound_sessions: BoundedSessions::default(),
-            outbound_sessions: BoundedSessions::default(),
+            outbound_consolidated_sessions: BoundedSessions::default(),
+            outbound_unconsolidated_sessions: BoundedSessions::default(),
         }
     }
 }
 
 impl<T> Sessions<T> {
     /// Method to get the sessions map by a session type
-    fn get_sessions_by_type(&mut self, session_type: SessionType) -> &mut BoundedSessions<T> {
+    fn get_sessions(
+        &mut self,
+        session_type: SessionType,
+        status: SessionStatus,
+    ) -> &mut BoundedSessions<T> {
         match session_type {
             SessionType::Inbound => &mut self.inbound_sessions,
-            SessionType::Outbound => &mut self.outbound_sessions,
+            SessionType::Outbound => match status {
+                SessionStatus::Unconsolidated => &mut self.outbound_unconsolidated_sessions,
+                SessionStatus::Consolidated => &mut self.outbound_consolidated_sessions,
+            },
         }
     }
     /// Method to set the server address
-    pub fn set_server_address(&mut self, server_adress: SocketAddr) {
-        self.server_address = Some(server_adress);
+    pub fn set_server_address(&mut self, server_address: SocketAddr) {
+        self.server_address = Some(server_address);
     }
     /// Method to set the sessions limits
-    pub fn set_limits(&mut self, inbound_limit: u16, outbound_limit: u16) {
+    pub fn set_limits(&mut self, inbound_limit: u16, outbound_consolidated_limit: u16) {
         self.inbound_sessions.set_limit(inbound_limit);
-        self.outbound_sessions.set_limit(outbound_limit);
+        self.outbound_consolidated_sessions
+            .set_limit(outbound_consolidated_limit);
     }
     /// Method to check if a socket address is eligible as outbound peer
     pub fn is_outbound_address_eligible(&self, candidate_addr: SocketAddr) -> bool {
-        // Check if address is already used as outbound session
-        let is_outbound = self
-            .outbound_sessions
+        // Check if address is already used as outbound session (consolidated or unconsolidated)
+        let is_outbound_consolidated = self
+            .outbound_consolidated_sessions
+            .collection
+            .contains_key(&candidate_addr);
+        let is_outbound_unconsolidated = self
+            .outbound_unconsolidated_sessions
             .collection
             .contains_key(&candidate_addr);
+
         // Check if address is the server address
         let is_server =
             self.server_address.is_some() && self.server_address.unwrap() == candidate_addr;
```

Comment:
Maybe for the sake of uniformity and avoiding wild unwraps:
```suggestion
            self.server_address.map(|address| address == candidate_addr).unwrap_or(false);
```